addons/frag/functions/fnc_dev_debugAmmo.sqf

```diff
@@ -1,81 +1,103 @@
-        private _force = getNumber (_ammoConfig >> QGVAR(force));
-        private _fragPower = getNumber (_ammoConfig >> "indirecthit") * (sqrt ((getNumber (_ammoConfig >> "indirectHitRange"))));
-
-        private _shouldAdd = (_skip == 0) && {(_force == 1) || {_explosive > 0.5 && {_indirectRange >= 4.5} && {_fragPower >= 35}}};
+        private _ammoConfig = (configFile >> "cfgAmmo" >> _ammo);
+        private _shoulFrag = [_ammo] call FUNC(shouldFrag);
 
-        if (_shouldAdd) then {
-            if (_debugForce && {((getNumber(_ammoConfig >> "hit")) < 5) || {_fragPower < 10}}) then {
-                diag_log text format ["Ammo [%1] from Mag [%2] - Weak but will still frag!", _ammo, configName _x];
-                diag_log text format [" - _force=%1,_fragPower=%2", _force, _fragPower];
-            };
+        if (_shoulFrag || _logAll) then {
 
             private _warn = false;
-
-            _fragTypes = getArray (_ammoConfig >> QGVAR(CLASSES));
+            private _fragTypes = getArray (_ammoConfig >> QGVAR(CLASSES));
             if (_fragTypes isEqualTo []) then {_warn = true;};
-            _c = getNumber(_ammoConfig >> QGVAR(CHARGE));
+            private _c = getNumber(_ammoConfig >> QGVAR(CHARGE));
             if (_c == 0) then {_warn = true;};
-            _m = getNumber(_ammoConfig >> QGVAR(METAL));
+            private _m = getNumber(_ammoConfig >> QGVAR(METAL));
             if (_m == 0) then {_warn = true;};
-            _k = getNumber(_ammoConfig >> QGVAR(GURNEY_K));
+            private _k = getNumber(_ammoConfig >> QGVAR(GURNEY_K));
             if (_k == 0) then {_warn = true;};
-            _gC = getNumber(_ammoConfig >> QGVAR(GURNEY_C));
+            private _gC = getNumber(_ammoConfig >> QGVAR(GURNEY_C));
             if (_gC == 0) then {_warn = true;};
+            private _fragCount = getNumber (_ammoConfig >> QGVAR(fragCount));
+            if (_fragCount == 0) then {_fragCount = 200; _warn = true;};
 
-            if (_debugMissing && {_warn}) then {
-                diag_log text format ["Ammo [%1] from Mag [%2] MISSING frag configs:", _ammo, configName _x];
-                diag_log text format [" - _c=%1,_m=%2,_k=%3,_gC=%4,_fragTypes=%5", _c, _m, _k, _gC, _fragTypes];
-            };
-        } else {
-            if ((_fragPower > _debugSkippedFragPower) && {isArray (_ammoConfig >> QGVAR(CLASSES))}) then {
-                diag_log text format ["Ammo [%1] from Mag [%2] has frag configs but will NOT frag:", _ammo, configName _x];
-                diag_log text format ["- skip=%1,explosive=%2,indirectHitRange=%3,force=%4,fragPower=%5", _skip, _explosive, _indirectRange, _force, _fragPower];
+            if (_warn) then {
+                INC(_printCount);
+                if (_csvFormat) then {
+                    diag_log text format ["%7,%1,%2,%3,%4,%5,%6,%8", _c, _m, _k, _gC, _fragTypes, _fragCount, _ammo, [_ammoConfig, true]  call BIS_fnc_returnParents];
```

Comment:
```suggestion
                    diag_log text format ["%7,%1,%2,%3,%4,%5,%6,%8", _c, _m, _k, _gC, _fragTypes, _fragCount, _ammo, [_ammoConfig, true] call BIS_fnc_returnParents];
```

Reply:
Fixed in https://github.com/acemod/ACE3/pull/9728/commits/054b9cda2367f481e66677af914870008237adcf